lib/matplotlib/quiver.py

```diff
@@ -251,47 +256,154 @@ def __init__(self, Q, X, Y, U, label,
+        """
+        Set the *x* position of the QuiverKey.
+
+        Parameters
+        ----------
+        x : float
+            The *x* location of the key.
+        """
+        self._text.set_x(x)
+        self.stale = True
+
+    X = property(get_x, set_x)
+
+    def get_y(self):
+        """Return the *y* position of the QuiverKey."""
+        return self._text.get_position()[1]
+
+    def set_y(self, y):
+        """
+        Set the *y* position of the QuiverKey.
+
+        Parameters
+        ----------
+        y : float
+            The *y* location of the key.
+        """
+        self._text.set_y(y)
+        self.stale = True
+
+    Y = property(get_y, set_y)
+
+    def get_position(self):
+        """Return the (x, y) position of the QuiverKey."""
+        return self._text.get_position()
+
+    def set_position(self, xy):
+        """
+        Set the position of the QuiverKey.
+
+        Parameters
+        ----------
+        xy : (float, float)
+            The (*x*, *y*) position of the QuiverKey.
+        """
+        self._text.set_position(xy)
+        self.stale = True
+
+    def get_label(self):
+        """Return the label string."""
+        return self._text.get_text()
+
+    def set_label(self, text):
+        """Set the label string."""
+        self._text.set_text(text)
+        self.stale = True
+
+    label = property(get_label, set_label, doc="The label string.")
+
+    def get_labelcolor(self):
+        """Return the label color."""
+        return self._text.get_color()
+
+    def set_labelcolor(self, labelcolor):
+        """Set the label color."""
+        self._text.set_color(labelcolor)
+        self.stale = True
+
+    labelcolor = property(get_labelcolor, set_labelcolor,
+                          doc="The label color.")
+
+    def get_labelpos(self):
+        """Return the label position."""
+        return self._labelpos
+
+    def set_labelpos(self, labelpos):
+        """
+        Set the label position.
+
+        Parameters
+        ----------
+        labelpos : {'N', 'S', 'E', 'W'}
+            Position the label above, below, to the right, to the left of the
+            arrow, respectively.
+        """
+        _api.check_in_list(['N', 'S', 'E', 'W'], labelpos=labelpos)
+        self._labelpos = labelpos
+        self._text.set_horizontalalignment(self.halign[labelpos])
+        self._text.set_verticalalignment(self.valign[labelpos])
+        self._update_text_transform()
+        self._initialized = False
+        self.stale = True
+
+    labelpos = property(get_labelpos, set_labelpos, doc="The label position.")
+
+    def get_labelsep(self):
+        """Return the distance between the arrow and label in inches."""
+        return self._labelsep_inches
+
+    def set_labelsep(self, labelsep):
+        """Set the distance between the arrow and label in inches."""
+        self._labelsep_inches = labelsep
+        self._update_text_transform()
+        self.stale = True
+
+    @property
```

Comment:
consider deprecating this, given the potential source of confusion?

Reply:
Agree.